src/app/elements/elements.component.ts

```diff
@@ -19,7 +19,8 @@ export class ElementsComponent implements OnInit {
 
   public components = new Map<string, ComponentRef<any>>();
   private indexNumber = 0;
-  public compts: any[] = []
+  public isDialogOpen: boolean = false;
+
   private componentTypes: any = {
```

Comment:
Suggestion: create a Map that contains the component types already associated to the corresponding component. Then in `getComponentType` you just use the get function of Map.
```typescript
private componentTypes: Map<string, Type<any>> = new Map([
    ['textElementComponent', TextElementComponent], ...
]);
```